int64 num
int64 swap_id1
int64 swap_id2
int64 extra_num
